### PR TITLE
Translate where designator

### LINKED ISSUE #50: src/generics 5
1. [where.md](https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/generics/where.md)
2. [designators.md](https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/macros/designators.md)

### PR DESCRIPTION
**Pull request by benhaq**
closes #50 

Translate generic/where and macro/designators

Depends on: #42 

- [x] [where.md](https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/generics/where.md)
- [x] [designators.md](https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/macros/designators.md)

### CODE REVIEW

**Comment by benhaq** [comment]
@HongThaiPham 

Phần dịch về 
[input_parameters, output_parameters](https://github.com/nebula-labs/rust-by-example-vn/pull/56/commits/f616d884fd68691eab03c9905338924bda4d9154) anh đừng quan tâm ạ vì nó nằm ở PR chưa được merge: https://github.com/nebula-labs/rust-by-example-vn/pull/42.
Anh chỉ cần check dùm em commit này thôi ạ:
[translate where.md and designators.md](https://github.com/nebula-labs/rust-by-example-vn/pull/56/commits/640566cc05ac336e867d23f3b800e035d095504e)

File: src/generics/where.md [hunk 1]
```diff
@@ -1 +1,49 @@
-# Where clauses
+# Mệnh đề where
+
+Một bound (ràng buộc) cũng có thể được định nghĩa bằng cách sử dụng mệnh đề `where` ngay trước dấu `{`, thay vì phải định nghĩa tại ngay lần đề cập đầu tiên của một kiểu.
+Ngoài ra, các mệnh đề `where` có thể áp dụng bound cho các kiểu tuỳ ý, chứ không chỉ riêng cho các tham số kiểu (type parameters).
+
+Việc sử dụng mệnh đề `where` sẽ hữu dụng khi rơi vào những trường hợp sau:
+
+- Khi muốn rõ ràng hơn bằng cách tách phần chỉ định các kiểu generic ra riêng và chỉ định bound ra riên
```
Comment (HongThaiPham) [inline src/generics/where.md]:
```suggestion
- Khi chỉ định rõ các kiểu dữ liệu generic và bound của chúng một cách riêng biệt.
```

File: src/macros/designators.md [hunk 2]
```diff
@@ -1 +1,60 @@
-# Designators
+# Định danh (designators)
+
+Các đối số của một macro sẽ được đánh dấu tiền tố bằng dấu đô la `$` và được gán kiểu với một định danh (designator).
```
Comment (HongThaiPham) [inline src/macros/designators.md]:
```suggestion
Các đối số của một macro sẽ được đánh dấu bằng tiền tố dấu đô la `$` và chú thích với kiểu bằng designator.
```

File: src/macros/designators.md [hunk 3]
```diff
@@ -1 +1,60 @@
...
+
+```rust,editable
+macro_rules! create_function {
+    // Macro nay nhận đối số với định danh là `ident` và
```
Comment (HongThaiPham) [inline src/macros/designators.md]:
```suggestion
    // Macro nay nhận đối số với designator là `ident` và
```

File: src/macros/designators.md [hunk 4]
```diff
@@ -1 +1,60 @@
...
+    // tạo ra một hàm có tên là `$func_name`
+    // Định danh `ident` được dùng cho tên của biến hoặc là tên của hàm
```
Comment (HongThaiPham) [inline src/macros/designators.md]:
```suggestion
    // Designator `ident` được dùng cho tên của biến hoặc là tên của hàm
```

File: src/macros/designators.md [hunk 5]
```diff
@@ -1 +1,60 @@
...
+        }
+    };
+}
+// Tạo ra một hàm tên là `foo` và `bar` với macro được định nghĩa ở trên.
+create_function!(foo);
+create_function!(bar);
+macro_rules! print_result {
+    // Macro này sẽ nhận vào một biểu thức (expression) với kiểu là `expr` và sẽ in
+    // biểu thức đó ra dưới dạng chuỗi string và kèm theo đó là kết quả của nó
+    // Định danh `expr` được dùng cho các biểu thức (expressions)
```
Comment (HongThaiPham) [inline src/macros/designators.md]:
```suggestion
    // Designator `expr` được dùng cho các biểu thức (expressions)
```

File: src/macros/designators.md [hunk 6]
```diff
@@ -1 +1,60 @@
...
+    print_result!(1u32 + 1);
+    // Nhắc lại là các blocks cũng là các biểu thức (expressions)
+    print_result!({
+        let x = 1u32;
+        x * x + 2 * x - 1
+    });
+}
+```
+
+Dưới đây là một vài định danh có sẵn:
```
Comment (HongThaiPham) [inline src/macros/designators.md]:
```suggestion
Dưới đây là một vài designator có sẵn:
```

File: src/macros/designators.md [hunk 7]
```diff
@@ -1 +1,60 @@
...
+- `item`
+- `literal` được dùng cho các hằng số literal (literal constants)
+- `pat` (_pattern_)
+- `path`
+- `stmt` (_statement_)
+- `tt` (_token tree_)
+- `ty` (_type_)
+- `vis` (_visibility qualifier_)
+
+Nếu muốn coi một danh sách đầy đủ của các định danh, hãy xem ở đây [Rust Reference]
```
Comment (HongThaiPham) [inline src/macros/designators.md]:
```suggestion
Nếu muốn coi một danh sách đầy đủ của các designator, hãy xem ở đây [Rust Reference]
```

**Review by HongThaiPham — APPROVED** [review]

**Pull request merged**